src/adm/filter.rs

```diff
@@ -237,3 +251,48 @@ impl AdmFilter {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
It's pretty clear that this bug doesn't exist now, but I think we should add a test that checks that an allowed host of "foo.com" doesn't allow "evilfoo.com". ie: it doesn't match when considering domain parts, but would match when (wrongly) considering string comparisons. It would be embarrassing for a bug like that to sneak in.